### ISSUE TITLE
Unexpected failure with single line else expression what appears to be correct formatting in 0.40

### ISSUE THREAD

**Issue by ZacSweers** [pinterest/ktlint#1047]
## Expected Behavior
This is correctly formatted

## Observed Behavior
It is reported as failure.

## Steps to Reproduce

Pasting exactly as-is in source for details

```kotlin
  override fun getActionScope(channelId: String?, threadTs: String?, isReadOnly: Boolean): AppActionScope {
    return if (channelId.isNullOrEmpty()) {
      AppActionNoChannelScope
    } else if (isReadOnly) {
      if (threadTs.isNullOrEmpty()) AppActionReadOnlyChannelScope(channelId) else AppActionReadOnlyChannelThreadScope(
        channelId,
        threadTs
      // The next line if the line it doesn't like
      )
    } else {
      if (threadTs.isNullOrEmpty()) AppActionChannelScope(channelId) else AppActionThreadScope(
        channelId,
        threadTs
      )
    }
  }
```

This results in this error:

```
SlackGlobalActionsHelper.kt:112:7: Unexpected indentation (expected 8, actual 6)
```

## Your Environment

* Version of ktlint used: 0.40.0

This is our editorconfig:
```
[*.{kt,kts}]
indent_size=2
continuation_indent_size=2
disabled_rules=final-newline
```
